logic/Rule.java

```diff
@@ -63,7 +74,7 @@ private Rule(LogicManager logicManager, RuleStructure structure) {
     }
 
     private Rule(GraphManager graphMgr, ConceptManager conceptMgr, LogicManager logicManager, String label,
-                 graql.lang.pattern.Conjunction<? extends Pattern> when, ThingVariable<?> then) {
+                 graql.lang.pattern.Conjunction<? extends Pattern> when, graql.lang.pattern.variable.ThingVariable<?> then) {
```

Comment:
I think that when we use both analogous objects from both Grakn and Graql, we should use the "long import" for the Object that is foreign to the current repo. Therefore, I've changed all `ThingVariable<?>` to `graql.lang.pattern.variable.ThingVariable<?>` in this PR.